### PR TITLE
Prevent caching of admin pages

### PR DESCRIPTION
**Pull request by gberaudo**
See https://jira.camptocamp.com/browse/GSGMF-468.

### CODE REVIEW

**Comment by arnaud-morvan** [comment]
Seems the README is not uptodate, noone build docker images for admin interface anymore.
We just run it in dev mode with make serve.

**Comment by gberaudo** [comment]
Thanks @arnaud-morvan for the information, it helped me in my quest to run the admin project locally.

I updated the readme.
@arnaud-morvan, could you please review? (@sbrunner is ill).

File: admin/c2cgeoportal_admin/__init__.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def main(_, **settings):
     config.add_subscriber(add_renderer_globals, BeforeRender)
     config.add_subscriber(add_localizer, NewRequest)
 
+    config.add_tween('c2cgeoportal_admin.cache_tween.CacheTween')
```
[the remark this answers is not in the document]
Comment (gberaudo) [reply admin/c2cgeoportal_admin/__init__.py]: If I remove this line the no-cache header is no more present in the responses.

File: admin/README.md [hunk 2]
```diff
@@ -41,8 +18,8 @@ sudo -u postgres psql -d $DATABASE -c "CREATE EXTENSION postgis;"
 
 Optionally update sqlachemy.url in development.ini or production.ini then:
 ```
-use common/testing/initialized.py to create the database
-use demo-dump.sql to create and populate the database (demo data)
+admin/.build/venv/bin/python3 commons/c2cgeoportal_commons/testing/initializedb.py # to create the database
+sudo -u postgres psql -d c2cgeoportal -f ./docker/demo-dump.sql # to create and populate the database (demo data)
```
[the remark this answers is not in the document]
Comment (gberaudo) [reply admin/README.md]: What do you mean? Where should I put the "or"?

[the remark this answers is not in the document]
Comment (arnaud-morvan) [reply admin/README.md]: If I'm not wrong, we should only execute one of this commands to initialize the database.
The dump already contains the tables creation.

File: admin/c2cgeoportal_admin/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arnaud-morvan) [reply admin/c2cgeoportal_admin/__init__.py]: Yes but this apply on all c2cgeoportal views, I'm not sure this is expected.

File: admin/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gberaudo) [reply admin/README.md]: Ah, OK. I did both. I will update the wording.

**Comment by gberaudo** [comment]
> Seems your tween apply to all c2cgeoportal, not only admin

Hmm, OK will look at this and add a prefix filter if needed.

**Comment by arnaud-morvan** [comment]
Note that the prefix is given here https://github.com/camptocamp/c2cgeoportal/blob/master/geoportal/c2cgeoportal_geoportal/__init__.py#L241

**Comment by arnaud-morvan** [comment]
I do not know it you can access the prefix in admin.includeme, but it should be possible.
Or the caching could be handled somewhere else, in a global tween for c2cgeoportal_geoportal ?

**Comment by gberaudo** [comment]
I looked at https://docs.pylonsproject.org/projects/pyramid/en/latest/_modules/pyramid/config.html#Configurator.include and it seems the `route_prefix` ends being a property of the config object. So I updated the PR (but did not test: I am only able to start the admin project at the moment).

Does it looks good to you?

**Comment by gberaudo** [comment]
Thanks @arnaud-morvan for the feedback. I fixed the typo and it should not be ready for merging.
Should I merge it today or wait for @sbrunner?

**Comment by arnaud-morvan** [comment]
I feel a little strange to handle caching specifically for admin part.
I do not know what is done on c2cgeoportal_geoportal concerning caching headers.
I can approve, but @sbrunner may have more comments about this.

File: admin/c2cgeoportal_admin/cache_tween.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+class CacheTween:
+
+    def __init__(self, handler, registry):
+        del registry
+        self.handler = handler
+
+    def __call__(self, request):
+        # Never cache admin pages
+        response = self.handler(request)
+        if route_prefix is None or request.path_info.startswith(route_prefix):
```
Comment (sbrunner) [inline admin/c2cgeoportal_admin/cache_tween.py]: Missing '/' before route_prefix...

Comment (gberaudo) [reply admin/c2cgeoportal_admin/cache_tween.py]: OK, fixed

**Comment by gberaudo** [comment]
Npm is broken, restarting the builds.

**Comment by gberaudo** [comment]
Done!

**Pull request merged**